### PR TITLE
7.x config query

### PR DESCRIPTION
**Pull request by nhart**
add config for hiding or showing child objects in parent collections

### CODE REVIEW

File: includes/admin.form.inc [hunk 1]
```diff
@@ -27,6 +27,13 @@ function islandora_compound_object_admin_form($form, &$form_state) {
     '#default_value' => variable_get('islandora_compound_object_thumbnail_child', TRUE),
   );
 
+  $form['islandora_compound_object_hide_child_in_collection'] = array(
```
Comment (willtp87) [inline includes/admin.form.inc]: We should avoid the use of the word collection here.

File: includes/admin.form.inc [hunk 2]
```diff
@@ -27,6 +27,13 @@ function islandora_compound_object_admin_form($form, &$form_state) {
...
+    '#type' => 'checkbox',
+    '#title' => t('Hide child objects in collection'),
+    '#description' => t('If checked, child objects will be hidden in their parent collections.'),
+    '#default_value' => variable_get('islandora_compound_object_hide_child_in_collection', TRUE),
```
Comment (willtp87) [inline includes/admin.form.inc]: Added var needs an uninstall.

File: includes/admin.form.inc (hunk 1 above)
Comment (nhart) [inline includes/admin.form.inc]: true. any suggestions?

**Comment by jordandukart** [comment]
Would likely be optimal to update the filter to Solr queries to respect this variable as well for consistency's sake: https://github.com/jordandukart/islandora_solution_pack_compound/blob/master/islandora_compound_object.module#L374-L382.

**Comment by jordandukart** [comment]
Also while you're here can you update this hook to use the non deprecated verison plx? https://github.com/jordandukart/islandora_solution_pack_compound/blob/master/islandora_compound_object.module#L183-L197 should now be `hook_islandora_xml_form_builder_form_associations()`.

**Comment by willtp87** [comment]
Can we make that a separate var for Solr?

**Comment by nhart** [comment]
I didn't wrap the solr bit in the var because the logic is already there if the field is empty.

File: islandora_compound_object.install [hunk 3]
```diff
@@ -25,6 +25,7 @@ function islandora_compound_object_uninstall() {
     'islandora_compound_object_compound_children',
     'islandora_compound_object_thumbnail_child',
     'islandora_compound_object_relationship',
+    'islandora_compound_object_hide_child_objects',
```
Comment (jordandukart) [inline islandora_compound_object.install]: Missing the two new fields as the name of this one changed.

File: includes/admin.form.inc [hunk 4]
```diff
@@ -27,12 +27,18 @@ function islandora_compound_object_admin_form($form, &$form_state) {
...
 
-  $form['islandora_compound_object_relationship'] = array(
-    '#type' => 'textfield',
-    '#title' => t('Child relationship predicate'),
-    '#description' => t('Changing this after objects have been created will break functionality. Should be part of info:fedora/fedora-system:def/relations-external#'),
-    '#default_value' => variable_get('islandora_compound_object_relationship', 'isConstituentOf'),
-    '#required' => TRUE,
+  $form['islandora_compound_object_hide_child_objects_ri'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Hide child objects in RI Results'),
```
Comment (jordandukart) [inline includes/admin.form.inc]: Lower r?

File: islandora_compound_object.module [hunk 5]
```diff
@@ -371,16 +371,17 @@ EOQ;
...
 /**
  * Implements hook_islandora_basic_collection_get_query_optionals().
  */
 function islandora_compound_object_islandora_basic_collection_get_query_optionals($type) {
-  if ($type == 'view') {
-    return format_string('?object <!uri!pred> ?compound', array(
-      '!uri' => FEDORA_RELS_EXT_URI,
-      '!pred' => variable_get('islandora_compound_object_relationship', 'isConstituentOf'),
-    ));
+  if (variable_get('islandora_compound_object_hide_child_objects', TRUE)) {
```
Comment (jordandukart) [inline islandora_compound_object.module]: This variable should be `islandora_compound_object_hide_child_objects_ri`.

File: islandora_compound_object.module [hunk 6]
```diff
@@ -388,8 +389,10 @@ function islandora_compound_object_islandora_basic_collection_get_query_optional
  * Implements hook_islandora_basic_collection_get_query_filters().
  */
 function islandora_compound_object_islandora_basic_collection_get_query_filters($type) {
-  if ($type == 'view') {
-    return '!bound(?compound)';
+  if (variable_get('islandora_compound_object_hide_child_objects', TRUE)) {
```
Comment (jordandukart) [inline islandora_compound_object.module]: This variable should be `islandora_compound_object_hide_child_objects_ri`.

**Pull request merged**